SYCL/ESIMD/api/functional/ctors/common.hpp

```diff
@@ -29,54 +29,50 @@ using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
 template <typename DataT>
 using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
 
-// Calls simd constructor in provided invocation context, which depends on the
-// TestCaseT type. TestCaseT is a struct, that should have call_simd_ctor method
-// that should return constructed object of simd class.
-// This function returns std::vector instance with the output data.
+// Function that verifies obtained test result.
 template <typename DataT, int NumElems, typename TestCaseT>
-auto call_simd(sycl::queue &queue, const shared_vector<DataT> &ref_data) {
-
-  shared_vector<DataT> result{NumElems, shared_allocator<DataT>{queue}};
-
-  queue.submit([&](sycl::handler &cgh) {
-    const auto ref = ref_data.data();
-    auto out = result.data();
+bool verify_test_result(const std::vector<DataT> &ref_data,
+                        const shared_vector<DataT> &result_data,
+                        const std::string &data_type) {
+  bool passed{true};
+
+  for (size_t i = 0; i < result_data.size(); ++i) {
+    if (!are_bitwise_equal(ref_data[i], result_data[i])) {
+      passed = false;
+      log::fail<NumElems>(
+          "Simd by " + TestCaseT::get_description() +
+              " failed, retrieved: " + std::to_string(result_data[i]) +
+              ", expected: " + std::to_string(ref_data[i]) +
+              ", at index: " + std::to_string(i),
+          data_type);
+    }
```

Comment:
Are all these changes about moving out the following lines?
```
  for (size_t i = 0; i < result_data.size(); ++i) {
    if (!are_bitwise_equal(ref_data[i], result_data[i])) {
```

The `log::fail` message will differ from test group to test group significantly. For example, conversion constructor would need to output both the source and destination types, the simd_mask or simd_view tests would have different template parameters - not to mention free functions. Or what if you have 2 vectors to compare for each case?

What do you think about moving out a `log::fail` message construction into a test description class instead? It may:

- have an abstract interface;
- result in some `log::fail(const ITestDescription&)` interface with no `NumElems` at all;
- provide a possibility to have generic `TestDescription: public ITestDescription` in a `ctors/common.hpp`
- moreover, you might even have `TestCaseT` either to be inherited from `TestDescription` or to have it encapsulated, if that's appropriate

Reply:
I've applied this comment in this commit e44cd0f653ddd3f6c73aa85230716eb71cad054e
I hope that I correctly understood you